packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -49,14 +49,19 @@ export interface VScopedSlotFragment extends BaseVNode {
     slotName: unknown;
 }
 
-export type VStaticElementData = Pick<VElementData, 'on' | 'ref'>;
+export interface VStaticPart {
+    readonly partId: number;
+    readonly data: VStaticPartData;
+    elm: Element | undefined;
+}
```

Comment:
Feedback from Dale: can we extract the `elm` and `data` from this interface into something shared with the normal `VNode`/`VElement` interface.

Reply:
I looked into this, but I think the types have sufficiently little overlap that it actually makes the TypeScript quite a bit more complex to DRY this out. I tried having some `HasData` and `HasElement` and `HasNode` "mixins" but I think this is actually more cumbersome than just copy-pasting the `elm: Element | undefined;` and `data: ...` props.